ape_solidity/compiler.py

```diff
@@ -361,6 +375,19 @@ def _get_settings_from_version_map(
             if solc_version >= Version("0.7.5") and config.via_ir is not None:
                 version_settings["viaIR"] = config.via_ir
 
+            if config.optimization_yul is not None:
+                if solc_version >= Version("0.8.22"):
+                    version_settings["optimizer"]["yul"] = config.optimization_yul
+                    if (
+                        config.optimization_steps is not None
+                    ):  # needs yul so is inside the yul check
+                        version_settings["optimizer"]["steps"] = config.optimization_steps
+                else:
+                    logger.warning(
+                        """Yul optimization and steps are not supported for this version of solc
+                     (Need >= 0.8.22)."""
+                    )
+
```

Comment:
I think this likely needs some changes up here https://github.com/ApeWorX/ape-solidity/pull/162/files#diff-de61acbec749a43c9ae1590d091b2924af42588fa689b9cef657a9e02f39e016R359

Maybe move it here to be:
```suggestion

else:
    version_settings["optimzer"].update({"enabled": config.optimize, "runs": config.optimization_runs})

```

(Also remove the setting above it and change the type hint for `version_settings` to just `dict`